Sources/MapboxMaps/Foundation/MapView+Lifecycle.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import UIKit
+
+@available(iOSApplicationExtension, unavailable)
+extension MapView {
+    private static let UIApplicationSceneManifestKey = "UIApplicationSceneManifest"
+
+    internal func subscribeToMemoryWarningNotification() {
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(didReceiveMemoryWarning),
+                                               name: UIApplication.didReceiveMemoryWarningNotification,
+                                               object: nil)
+    }
+
+    internal func subscribeToLifecycleNotifications() {
+        if Bundle.main.object(forInfoDictionaryKey: MapView.UIApplicationSceneManifestKey) != nil {
+            if #available(iOS 13.0, *) {
```

Comment:
What happens if there's an app that uses scenes in iOS 13+ and falls back to older tech when running on older iOS versions?

Reply:
Well spotted! Fixed it.